bellman/src/domain/linear.rs

```diff
@@ -0,0 +1,163 @@
+use std::ops::{Add, Sub};
+use ff::{ScalarEngine, Field};
+
+/// Represents a variable in our constraint system.
+#[derive(Copy, Clone, Debug)]
+pub struct Variable(Index);
+
+impl Variable {
+    /// This constructs a variable with an arbitrary index.
+    /// Circuit implementations are not recommended to use this.
+    pub fn new_unchecked(idx: Index) -> Variable {
+        Variable(idx)
+    }
+
+    /// This returns the index underlying the variable.
+    /// Circuit implementations are not recommended to use this.
```

Comment:
Do not need to repeat, just define it once in docs, otherwise it is just clutter.

Reply:
I’m leaving most of their comments in until it’s been moved down into the code, otherwise I’m concerned it might contain a nugget of information that we need to keep in mind. 

The repeat comment indicates I might be able to unify the behaviour under an interface, where is it repeating?

Reply:
Agreed, sounds like a good plan of action